modules/exadsBidAdapter.js

```diff
@@ -0,0 +1,433 @@
+import * as utils from '../src/utils.js';
+import { BANNER, NATIVE, VIDEO } from '../src/mediaTypes.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+
+const BIDDER = 'exadsadserver';
+
+const PARTNERS = {
+  RTB_2_4: 'rtb_2_4'
+};
+
+const envParams = {
+  lang: '',
+  userAgent: '',
+  osName: '',
+  page: '',
+  domain: '',
+  language: '',
+  userId: ''
+};
+
+const adPartnerHandlers = {
+  [PARTNERS.RTB_2_4]: {
+    request: handleReqRTB2Dot4,
+    response: handleResRTB2Dot4,
+    validation: handleValidRTB2Dot4,
+  }
+};
+
+function handleReqRTB2Dot4(bid, endpointUrl, validBidRequests, bidderRequest) {
+  utils.logInfo(`Calling endpoint for rtb_2_4:`, endpointUrl);
+  const gdprConsent = getGdprConsentChoice(bidderRequest);
+
+  // Make a dynamic bid request to the ad partner's endpoint
+  let bidRequestData = {
```

Comment:
I let @kristijanexads providing his input here, but we should be sure that we gather everything we could use from the validBidRequests or bidderRequest. I have in mind as an example:
- params.publisher
- params.bcat
- params.badv
- params.test
- bidderRequest.timeout
- etc.

Reply:
@broucz somewhat agree, but this is publisher side, and if we accept those (but we don't do anything about it) then they might also complain. 
But from the ones you mentioned above, `params.bcat` and `params.badv` are already in development, so those should be added to the request.

Reply:
although we need to add DSA to the header bidding request.

Reply:
Thanks for your input @kristijanexads, it makes sense.

@giuseppe-exads If you're not already on it, please be sure to follow up on Kris suggestion to add `bcat` and `badv` support to the request (DSA is already supported). Thanks!

Reply:
just added bcat and badv.